src/3box.js

```diff
@@ -731,13 +752,23 @@ async function initIPFS (ipfs, iframeStore, ipfsOptions) {
   } else {
     // await iframeLoadedPromise
     // return ipfsProxy
+    let ipfsRepo
+    if (!ipfsOptions) {
+      ipfsRepo = initIPFSRepo()
+      ipfsOptions = Object.assign(IPFS_OPTIONS, { repo: ipfsRepo.repo })
+    }
     return new Promise((resolve, reject) => {
-      ipfs = new IPFS(ipfsOptions || IPFS_OPTIONS)
+      ipfs = new IPFS(ipfsOptions)
       ipfs.on('error', error => {
         console.error(error)
         reject(error)
       })
-      ipfs.on('ready', () => resolve(ipfs))
+      ipfs.on('ready', () => {
+        resolve(ipfs)
+        if (ipfsRepo && window && window.indexedDB) {
+          window.indexedDB.deleteDatabase(ipfsRepo.rootPath)
```

Comment:
I'm don't understand this. Why do we delete the db right after ipfs in initialized? Wouldn't the db be used in this session?

Reply:
yeah the db is used, this is blocked until the process (ipfs repo) lets go of its lock (or open status) then it is deleted. So it still get deleted even after the window is abruptly closed

Reply:
This works nice for indexeddb (browser env) but something more stable for node would also handle the filesystem. In that case it would have look to delete older key repos on start instead of on close

Reply:
I don't think this would be much of an issue in node though since we don't have multiple tabs :P Maybe just don't randomize when in node?

Reply:
yeah that's one option, probs simplest at moment. But would be nice to get one logical instance of ipfs per instance of 3box, if running tests or multiple instances in one directory. But yeah I think should skip worrying about that for now 👍